dev/com.ibm.ws.security.openidconnect.clients.common/src/com/ibm/ws/security/openidconnect/clients/common/OIDCClientAuthenticatorUtil.java

```diff
@@ -233,9 +235,14 @@ void getUserInfo(ConvergedClientConfig clientConfig, Hashtable<String, String> r
         boolean needHttps = clientConfig.getUserInfoEndpointUrl().toLowerCase().startsWith("https");
         SSLSocketFactory sslSocketFactory = null;
         try {
-            sslSocketFactory = new OidcClientHttpUtil().getSSLSocketFactory(clientConfig, sslSupport, false, needHttps);
+            sslSocketFactory = new OidcClientHttpUtil().getSSLSocketFactory(clientConfig.getSSLConfigurationName(), sslSupport);
+            if (sslSocketFactory == null && needHttps) {
+                Tr.error(tc, "OIDC_CLIENT_HTTPS_WITH_SSLCONTEXT_NULL", new Object[] { "Null ssl socket factory", clientConfig.getClientId() });
+            }
```

Comment:
The `SecuritySSLUtils.getSSLSocketFactory()` method that the `OidcClientHttpUtil` class is calling will always throw an exception if the `SSLSocketFactory` is null, so you can move this check to the catch block for the `NoSSLSocketFactoryException`.
```suggestion
            sslSocketFactory = new OidcClientHttpUtil().getSSLSocketFactory(clientConfig.getSSLConfigurationName(), sslSupport);
```